plugins/package-managers/swiftpm/src/main/kotlin/SwiftPm.kt

```diff
@@ -164,3 +169,72 @@ class SwiftPm(
         )
     }
 }
+
+internal val SwiftPackage.Dependency.id: Identifier
+    get() = Identifier(
+        type = PACKAGE_TYPE,
+        namespace = "",
+        name = getCanonicalName(repositoryUrl),
+        version = version
+    )
+
+internal fun SwiftPackage.Dependency.toPackage(): Package {
+    val vcsInfoFromUrl = VcsHost.parseUrl(repositoryUrl)
+    val vcsInfo = vcsInfoFromUrl.takeUnless { it.revision.isBlank() } ?: vcsInfoFromUrl.copy(revision = version)
```

Comment:
Not introduced in this commit, but is there something sepcial in SwiftPM that allows us to set the revision to the version? We don't do this for other package managers and instead rely on the downloader to find a matching tag.

Reply:
Yes, there is something special. The implementation using `swift package show-dependencies` involves resolving the revisions. These resolved revisions are present in the lockfile. `SwifPM` operates directly on `Git` repository, similar as `GoMod` does. So, for these two package managers, revisions get resolved during building of the dependency tree.

Note: My next PR on top of this, will change the implementation exactly on this topic. I want to always set the resolved revision as revision. I'll do that obtaining these from the lockfile, which either already exists or is created when obtaining the dependencies.

Reply:
Ok, but this doesn't explain why the `version` is used as fallback revision.

Reply:
True, using the version as fallback is wrong IMO. 
I'll fix this. Next PR will improve setting the version string, and the revision.